athenz/src/main/java/com/linecorp/armeria/server/athenz/AthenzServiceBuilder.java

```diff
@@ -44,26 +46,47 @@ public final class AthenzServiceBuilder extends AbstractAthenzServiceBuilder<Ath
     private List<TokenType> tokenTypes = DEFAULT_TOKEN_TYPES;
 
     @Nullable
-    private String athenzResource;
+    private AthenzResourceProvider athenzResourceProvider;
+
     @Nullable
     private String athenzAction;
 
+    @Nullable
+    private String resourceTagValue;
+
     AthenzServiceBuilder(ZtsBaseClient ztsBaseClient) {
         super(ztsBaseClient);
     }
 
     /**
      * Sets the Athenz resource to check access permissions against.
      *
-     * <p><strong>Mandatory:</strong> This field must be set before calling {@link #newDecorator()}.
+     * <p><strong>Mandatory:</strong> Either this or {@link #resourceProvider(AthenzResourceProvider, String)} must be set before calling {@link #newDecorator()}.</p>
      */
     public AthenzServiceBuilder resource(String athenzResource) {
         requireNonNull(athenzResource, "athenzResource");
         checkArgument(!athenzResource.isEmpty(), "athenzResource must not be empty");
-        this.athenzResource = athenzResource;
+        athenzResourceProvider = (ctx, req) -> CompletableFuture.completedFuture(athenzResource);
+        resourceTagValue = athenzResource;
+        return this;
+    }
+
+
+    /**
+     * Sets the {@link AthenzResourceProvider} used to resolve the Athenz resource dynamically for each request.
+     *
+     * <p><strong>Mandatory:</strong> Either this or {@link #resource(String)} must be set before calling {@link #newDecorator()}.</p>
```

Comment:
style) This line exceeds our max line length.
https://github.com/line/armeria/blob/7148df07f3a7fe7cb4640fb1ca761cbaca8b347f/settings/checkstyle/checkstyle.xml#L26-L27

Reply:
I've applied the LY OSS code style to match Armeria's formatting convention